flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/BuiltInFunctionTestBase.java

```diff
@@ -342,9 +341,19 @@ TestSpec testResult(
                 List<Object> result,
                 List<AbstractDataType<?>> tableApiDataType,
                 List<AbstractDataType<?>> sqlDataType) {
-            testItems.add(new TableApiResultTestItem(expression, result, tableApiDataType));
-            testItems.add(
-                    new SqlResultTestItem(String.join(",", sqlExpression), result, sqlDataType));
+
+            expression = expression.stream().filter(Objects::nonNull).collect(Collectors.toList());
```

Comment:
I'm not a big fan of this, imho it would more clear to use directly the `testSqlResult()` & `testSqlValidationError()` directly and not go through the `ResultSpec` Pojo for this test. What do you think?

Reply:
got it. Thanks for your reminder :)